synapse/storage/database.py

```diff
@@ -390,14 +398,20 @@ def execute_values(
         from psycopg2.extras import execute_values
 
         return self._do_execute(
+            # TODO: is it safe for values to be Iterable[Iterable[Any]] here?
+            # https://www.psycopg.org/docs/extras.html?highlight=execute_batch#psycopg2.extras.execute_values says values should be Sequence[Sequence]
             lambda the_sql: execute_values(self.txn, the_sql, values, fetch=fetch),
             sql,
         )
 
-    def execute(self, sql: str, *args: Any) -> None:
-        self._do_execute(self.txn.execute, sql, *args)
+    def execute(self, sql: str, parameters: _Parameters = ()) -> None:
```

Comment:
Arg, why did we ever accept `*args` here if we only provide a single parameter?! I originally thought we were switching from providing parameters as `args` to now a single tuple, but seems this was just being overly generous... 😢

Reply:
I think we were trying to make a transparent wrapper. We also probably didn't have good annotations for the DBAPI2 at the time either

Reply:
DBAPI2 still has kind of weird annotations IIRC, but point taken!